packages/hoppscotch-common/src/newstore/collections.ts

```diff
@@ -494,6 +495,50 @@ const restCollectionDispatchers = defineDispatchers({
     }
   },
 
+  duplicateCollection(
+    { state }: RESTCollectionStoreType,
+    // `collectionSyncID` is used to sync the duplicated collection in `collections.sync.ts`
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    { path, collectionSyncID }: { path: string; collectionSyncID?: string }
+  ) {
+    const t = getI18n()
+
+    const newState = state
+
+    const indexPaths = path.split("/").map((x) => parseInt(x))
+
+    const isRootCollection = indexPaths.length === 1
+
+    const collection = navigateToFolderWithIndexPath(state, [...indexPaths])
+
+    if (collection) {
+      const name = `${collection.name} - ${t("action.duplicate")}`
+
+      const duplicatedCollection = {
+        ...cloneDeep(collection),
+        name,
+        ...(collection.id
+          ? { id: `${collection.id}-duplicate-collection` }
+          : {}),
```

Comment:
A `-duplicate-collection` suffix is added to the new collection following the ID from the original collection. This is to identify the newly created collection from the store and assign the backend-issued ID as received from the GQL subscription from the syncing context. This doesn't come into play at the moment since the action is disabled under personal workspace in SH. There are no side effects observed on `central`. It also applies to GQL.